core/integer/divide_spec.rb

```diff
@@ -1,7 +1,15 @@
 require File.expand_path('../../../spec_helper', __FILE__)
-require File.expand_path('../shared/arithmetic_exception_in_coerce', __FILE__)
+require File.expand_path('../shared/arithmetic_coerce', __FILE__)
 
-describe "Integer#/" do
-  it_behaves_like :integer_arithmetic_exception_in_coerce, :/
+ruby_version_is "2.4" do
+  describe "Integer#/" do
+    ruby_version_is "2.4"..."2.5" do
+      it_behaves_like :integer_arithmetic_coerce_rescue, :/
+    end
+
+    ruby_version_is "2.5" do
+      it_behaves_like :integer_arithmetic_coerce_not_rescue, :/
+    end
+  end
```

Comment:
Why did you extract the version guards outside of the shared spec in this commit?
It seems to make usages of the shared spec more verbose and fragile.
Are they problematic or unclear somehow?

Reply:
@eregon looks like I misunderstood your previous comment:

> Can the guards not be part of the shared example?

There are minor pros and cons so if you want I will revert this change